logfire/_internal/stack_info.py

```diff
@@ -65,3 +70,35 @@ def get_caller_stack_info(stack_offset: int = 3) -> StackInfo:
         return get_stack_info_from_frame(frame)
     except Exception:  # pragma: no cover
         return {}
+
+
+def get_user_stack_offset() -> int:
+    """Get the stack offset of the user code.
+
+    We want to skip the internal code, and third party code, and get the user code stack info.
+
+    Returns:
+        The stack offset of the user code.
+    """
+    stack_offset = 2
```

Comment:
@alexmojaki Do you know why the number 2 here works? I didn't check the stack closely...

Reply:
It doesn't seem like it *would* work as a default, and the pragma suggests that it hasn't been tested.

Reply:
The pragma on the exception you mean?

Reply:
Yes, since there's no exceptions happening (and I guess there generally wouldn't be) I think the final `stack_offset` is always greater than 2. Not sure if a lower starting point is needed.

Reply:
What should I do here?

Reply:
Nothing, I just realized I misunderstood the meaning of this number. I think this is fine.